### PR TITLE
OCPQE-18694: add workflow azure-ipi-fullyprivate-firewall

### PR DESCRIPTION
**Pull request by jinyunma**
automate test case: 
[OCP-53228 - [ipi-on-azure] Create fully private clusters using firewall as UserDefinedRouting outboundType](https://polarion.engineering.redhat.com/polarion/#/project/OSE/workitem?id=OCP-53228)

1. Add step `azure-provision-firewall` to provision firewall resources
2. Add workflow `cucushift-installer-rehearse-azure-ipi-fullyprivate-firewall`
3. Fix issues in `cucushift-installer-check-azure-vnet-tag`, get issue when several vnets created in same resource group, query vnet id with specific vnet name.

### CODE REVIEW

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jinyunma** [comment]
cc @jianlinliu @gpei to review

1. Due to the code in front of cloudfront.net is changed on different registries, restricted the job to be run on build05. Or any other thought?
```
redhat_fqdns_list="registry.build05.ci.openshift.org"
redhat_fqdns_list="${redhat_fqdns_list} d11wsypyfft1aa.cloudfront.net"
```
2. Currently, only add verification-tests jobs on 4.14+, do we need to add rehearse job with `ENABLE_FIREWALL_FULLLIST: yes`?

**Comment by gpei** [comment]
> cloudfront.net



> cc @jianlinliu @gpei to review
> 
> 1. Due to the code in front of cloudfront.net is changed on different registries, restricted the job to be run on build05. Or any other thought?
> 
> ```
> redhat_fqdns_list="registry.build05.ci.openshift.org"

Yes, it's ok to me to restrict the job to be run on build05. 

> redhat_fqdns_list="${redhat_fqdns_list} d11wsypyfft1aa.cloudfront.net"
> ```
If the prefix of cloudfront.net also various for build05 from time to time, I think we can also update it to `*.cloudfront.net`.

> 2. Currently, only add verification-tests jobs on 4.14+, do we need to add rehearse job with `ENABLE_FIREWALL_FULLLIST: yes`?
My opinion is we can limit this test as an installation test with minimal firewall list, for the team's E2E test, they may extend their testing case anytime so we may have to maintain the FULLLIST if new URL/website need to be opened.

**Comment by jinyunma** [comment]
> If the prefix of cloudfront.net also various for build05 from time to time, I think we can also update it to `*.cloudfront.net`

It seems to be fixed, ran job twice, same prefix.

> My opinion is we can limit this test as an installation test with minimal firewall list, for the team's E2E test, they may extend their testing case anytime so we may have to maintain the FULLLIST if new URL/website need to be opened.

agree to limit the test in verification-test.

**Comment by gpei** [comment]
> > If the prefix of cloudfront.net also various for build05 from time to time, I think we can also update it to `*.cloudfront.net`
> 
> It seems to be fixed, ran job twice, same prefix.

Or we can even set the URL for nightly registry as 
```
redhat_fqdns_list="*.ci.openshift.org"
redhat_fqdns_list="${redhat_fqdns_list} *.cloudfront.net"
```
In general, they're both internal Red Hat URLs used for nightly builds, it should be fine for our minimal firewall list test. I checked through the CI jobs, seems no other Azure job was restricted to build05 or any specific build farm, so that we won't be limited for where it runs, wdyt?

**Comment by jinyunma** [comment]
> In general, they're both internal Red Hat URLs used for nightly builds, it should be fine for our minimal firewall list test. I checked through the CI jobs, seems no other Azure job was restricted to build05 or any specific build farm, so that we won't be limited for where it runs, wdyt?

make sense, update to use wild url and remove hardcoded build farm cluster in job.

**Comment by jinyunma** [comment]
/pj-rehearse periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14

**Comment by jianlinliu** [comment]
> Fix issues in cucushift-installer-check-azure-vnet-tag, get issue when several vnets created in same resource group

To be curious, what case will create several vnets in one resource group?

File: ci-operator/step-registry/azure/provision/firewall/azure-provision-firewall-ref.metadata.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+	"path": "azure/provision/firewall/azure-provision-firewall-ref.yaml",
```
Comment (jianlinliu) [inline ci-operator/step-registry/azure/provision/firewall/azure-provision-firewall-ref.metadata.json:2]:
```suggestion
	"path": "azure/provision/vnet/firewall/azure-provision-firewall-ref.yaml",
```

**Comment by jianlinliu** [comment]
> > Fix issues in cucushift-installer-check-azure-vnet-tag, get issue when several vnets created in same resource group
> 
> To be curious, what case will create several vnets in one resource group?

Oh, got it, it is this case, it is creating a 2nd vent for firewall

File: ci-operator/step-registry/azure/provision/firewall/azure-provision-firewall-ref.metadata.json (hunk 1 above)
Comment (jinyunma) [reply ci-operator/step-registry/azure/provision/firewall/azure-provision-firewall-ref.metadata.json]: updated

**Comment by jinyunma** [comment]
job for change on nightly build related URL was passed, https://qe-private-deck-ci.apps.ci.l2s4.p1.openshiftapps.com/view/gs/qe-private-deck/pr-logs/pull/openshift_release/47471/rehearse-47471-periodic-ci-openshift-verification-tests-master-installation-nightly-4.15-azure-ipi-fullyprivate-firewall-f14/1745373815061352448
 
/pj-rehearse ack

**Pull request merged**